source/includes/inkling-reference/_algorithms.html.md

```diff
@@ -0,0 +1,196 @@
+
+```inkling-code
+schema GameState
+   Float32 position,
+   Float32 velocity,
+   Float32 angle,
+   Float32 rotation
+end
+
+schema Action
+   Float32 command
+end
+
+schema CartPoleConfig
+   Int8 episode_length,
+   UInt8 deque_size
+end
+
+simulator the_simulator(CartPoleConfig)
+   action (Action)
+   state (GameState)
+end
+
+concept balance is estimator
+   predicts (Action)
+   follows input(GameState)
+   feeds output
+
+end
+
+curriculum balance_curriculum
+   train balance
+   using algorithm TRPO
+      learning_rate => 5
+      network_size => 48
+   end
+
+   with simulator the_simulator
+   objective open_ai_gym_default_objective
+
+       lesson balancing
+           configure
+               constrain episode_length with Int8{-1},
+               constrain deque_size with UInt8{1}
+           until
+               maximize open_ai_gym_default_objective
+end
+```
+
+The algorithm clause is for advanced users who want to experiment changing machine learning algorithms and their training parameters. The clause provides syntax either within the curriculum statement to force a choice of algorithm for that curriculum, or it can be specified outside of a curriculum if used in multiple concepts (and therefore multiple lessons). It is also possible to override a global parameter inside of a curriculum if needed.
+
+### Valid Algorithm Identifiers
+
+* `DQN` (Deep Q Learning) - classification only
+* `DDPG` - estimation only
+* `DiscreteDDPG` - classification only
+* `QTable` - classification only
+* `TRPO` - estimation only
+
+
+### Shared Parameter Details
+
+The following parameters all apply to DQN, DDPG, Discrete DDPG, and TRPO. Additional parameter options are shown in subsequent tables below.
+
+| Parameter (Example use)      | Description |
+| -                            | -           |
+| `hidden_layer_size_descriptor` => [int, int, int, …] <br> (`hidden_layer_size_descriptor => [400, 300]` | An array of hidden layer: int is the hidden layer size. |
+| `hidden_layer_activation_descriptor` => [string, string, string, …] <br> (`hidden_layer_activation_descriptor => [“relu”, “tanh”]`) | An array of activation functions (corresponding to the hidden layer size descriptor): string is the activation function type. Strings supported: “linear”, “tanh”, “relu”, “logistic”, “softmax”, “elu”. (Using this should be combined with `hidden_layer_size_descriptor`, in the given example “relu” is the activation function for the first layer, and “tanh” for the second one.) |
+| `conv_layer_descriptor` => “int1xint2:int3:int4:int5; …” <br> (`conv_layer_descriptor => “8x8:4:4:32; 3x3:1:1:16”`) | A semicolon delimited string of convolutional network configuration: int1 is x size, int2 is y size, int3 is x stride, int4 is y stride, int5 is number of filters. |
+| `conv_compression_size_descriptor` => [int, int, int, ...] <br> (`hidden_layer_descriptor => [20, 10]`)| An array of convolutional compressed size: int is the hidden layer size. |
```

Comment:
copy paste leftovers: `hidden_layer_descriptor` instead of `conv_compression_size_descriptor`. Should add a note about what this is. e.g. `If specified, will add one or more fully connected layers after the convolutional layers to compress the representation before incorporating other features`.